backend/internal/system/utils/configutil.go

```diff
@@ -0,0 +1,153 @@
+package utils
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"text/template"
+)
+
+var (
+	// Pattern to match Go template variables like {{.Variable}}
+	varPattern = regexp.MustCompile(`\{\{\s*\.\s*([A-Za-z_][A-Za-z0-9_]*)\s*\}\}`)
+
+	// Pattern to match Go template range patterns like {{- range .ArrayVar}}
+	rangePattern = regexp.MustCompile(`\{\{-\s*range\s+\.\s*([A-Za-z_][A-Za-z0-9_]*)\s*\}\}`)
+
+	// Pattern to match Go template file path patterns
+	filePattern = regexp.MustCompile(`file://(?:"([^"]*)"|([^\s"]+))`)
+)
+
+// SubstituteFilePaths replaces file path placeholders in the given content with the actual file contents.
+//
+// Supported patterns:
+//  1. file://path/to/file - Unquoted file path (no spaces allowed)
+//  2. file://"path/with/spaces" - Quoted file path (spaces allowed)
+//  3. file:/relative/path - Relative file path (resolved against base directory)
+//
+// If a file cannot be read, an error is returned.
+func SubstituteFilePaths(content []byte, baseDir string) ([]byte, error) {
+	contentStr := string(content)
+	var readErr error
+
+	out := filePattern.ReplaceAllStringFunc(contentStr, func(match string) string {
+		sub := filePattern.FindStringSubmatch(match)
+		if len(sub) < 2 {
+			return match
+		}
+
+		// Group 1: quoted path (file://"path"), Group 2: unquoted path (file://path)
+		path := sub[1]
+		if path == "" {
+			path = sub[2]
+		}
+		if path == "" {
+			return ""
+		}
+
+		// Convert relative paths to absolute
+		if !filepath.IsAbs(path) {
+			path = filepath.Join(baseDir, path)
+		}
+
+		data, err := readFileContent(path)
+		if err != nil {
+			readErr = err
```

Comment:
This will only return the last error occurs, if there are multiple files referred and there are multiple errors occurred only the last error is returned.